omniscidb/ArrowStorage/ArrowStorage.cpp

```diff
@@ -203,7 +248,7 @@ void ArrowStorage::fetchFixedLenData(const TableData& table,
   // to get a proper slice.
   size_t elems = elem_size / arrow_elem_size;
   CHECK_GT(elems, (size_t)0);
-  auto data_to_fetch =
+  const auto& data_to_fetch =
```

Comment:
`Slice` creates a new `shared_ptr`, therefore you cannot use reference here and all other similar places.